### DISCUSSION TITLE
Queries on SP-API Capabilities

### DISCUSSION THREAD

**Discussion by Gopikrish25** [amzn/selling-partner-api-models#4181] [Q&A] [answered]
Hello,

Currently I have access to sandbox version of SP-API which returns static data incase of orders and it returns nothing for the reports part. I've requested PROD API access by filling in the questionare to get the API credentails for my selling account and as it is in progress of review, I would like to get some support from the SP-API team on the below queries so it can help me to develop an inteagration with my ERP system.

1. Does Orders API provide us the orders fulffilled via FBA program?
2. Is there an API to query return orders? [Or should we use reports for this?]
3. Let's assume like if I download an all order report from seller central and I would like to download the same exact report using SP-API will it return the same format and type? [Example file extension and data columns/headers etc..]

**NOTE: The seller account is part of the FBA program only.**

It would really help a lot if someone can answer my above queries. Thanks in advance!

Regards,

S Gopi Krishna
Consultant - D365 Business Applications

**Comment by jawheral** [comment, marked as answer]
Hi @Gopikrish25,

Thanks for reaching out.

1 - The Orders API provides you with all orders. You can filter results by `FulfillmentChannels` in this [operation](https://developer-docs.amazon.com/sp-api/docs/orders-api-v0-reference#get-ordersv0orders) or check the `FulfillmentChannel` in the order [response](https://developer-docs.amazon.com/sp-api/docs/orders-api-v0-reference#order).

2 - For returns, please check this return reports [section](https://developer-docs.amazon.com/sp-api/docs/report-type-values-returns) and the [FBA returns report](https://developer-docs.amazon.com/sp-api/docs/report-type-values-fba#fba-returns-report).

3 - That depends on the order report. For example XML and Flat File All order reports would be different. A flat-file would be mostly similar to an All-Order report downloaded from Seller Central. Please keep in mind that order reports columns can be customized using the column [picker](https://sellercentral.amazon.de/order-reports-and-feeds/column-picker?source=/order-reports-and-feeds/reports/). If the seller changes the columns on Seller Central, the report returned from the API will also include those.
You can check the report attributes for an all orders report [here](https://developer-docs.amazon.com/sp-api/docs/report-type-values-order#flat-file-orders-by-last-update-report).

SP-API also provides a Notifications API. the ORDER_CHANGE notification type is one example which is important to know as you work on your order workflow development. You can read more on this notification [here](https://developer-docs.amazon.com/sp-api/docs/tutorial-subscribe-to-order-change-notification).

Best,
Jawher
Selling Partner API Solutions Architect

**Reply by Gopikrish25** [reply]
Hey @jawheral ,

Thanks for your response it did help me and I have a last question, is my following assumption correct?

Reports cannot be generated using Sandbox API's.

**Reply by jawheral** [reply]
Hi @Gopikrish25,

There is a static sandbox for Reports API meaning that it can provide a static report but it doesn't generate reports.

Best,
Jawher
Selling Partner API Solutions Architect
